lib/rack_middleware/enforce_ssl.rb

```diff
@@ -0,0 +1,21 @@
+class EnforceSsl
+  def initialize(app)
+    @app = app
+  end
+
+  def call(env)
+    if ::APP_CONFIG.always_use_ssl.to_s == "true" && env["HTTPS"] == "off" # always_use_ssl can be string if it comes from ENV
+      req = ::Rack::Request.new(env)
+      port_s = req.port ? ":#{req.port}" : ""
+      redirect("https://#{req.host}#{port_s}#{req.fullpath}")
```

Comment:
Does req.fullpath include e.g. query string? I'm trying to think through if this can lose some data.